test/attn.py

```diff
@@ -0,0 +1,72 @@
+from torch import autograd
+import torch
+
+# Inherit from Function
+class AttentionFunction(autograd.Function):
+    @staticmethod
+    def attn(ctx, q, k, v):
```

Comment:
interesting that you would split this out, typically we'd just have this logic inside the forward